src/include/migraphx/op/reshape.hpp

```diff
@@ -46,14 +47,66 @@ struct reshape
     value attributes() const { return {{"require_std_shape", true}}; }
 
     std::string name() const { return "reshape"; }
-    shape compute_shape(std::vector<shape> inputs) const
+
+    shape dyn_compute_shape(shape s0) const
     {
-        check_shapes{inputs, *this}.has(1).standard();
+        auto dyn_dims       = s0.dyn_dims();
+        int not_fixed_index = -1;
+        // track number of fixed elements in input and output
+        std::size_t num_dims_ele = 1;
+        std::size_t num_dd_ele   = 1;
+        for(std::size_t i = 0; i < dyn_dims.size(); ++i)
+        {
+            if(dyn_dims[i].is_fixed())
+            {
+                num_dims_ele *= dims[i];
+                num_dd_ele *= dyn_dims[i].min;
+            }
+            else
+            {
+                if(not_fixed_index == -1)
+                {
+                    not_fixed_index = i;
+                }
+                else
+                {
+                    MIGRAPHX_THROW("Reshape: Only support one non-fixed dynamic_dimension");
+                }
+            }
+        }
+        if(num_dims_ele != num_dd_ele)
+        {
+            MIGRAPHX_THROW("Reshape: Number of fixed elements must match. Input: " +
+                           std::to_string(num_dd_ele) + " Output: " + std::to_string(num_dims_ele));
+        }
+        if(dims[not_fixed_index] != 0 and dims[not_fixed_index] != -1)
+        {
+            MIGRAPHX_THROW("Reshape: Non-fixed dynamic_dimension doesn't match with 0 or -1 "
+                           "output dimension");
+        }
+        // construct output dynamic shape from dims attribute
+        std::vector<shape::dynamic_dimension> output_dyn_dims = {};
+        for(std::size_t i = 0; i < dims.size(); ++i)
+        {
+            if(i == not_fixed_index)
+            {
+                output_dyn_dims.push_back(dyn_dims[not_fixed_index]);
+            }
+            else
+            {
+                std::size_t d = dims[i];
+                output_dyn_dims.push_back({d, d, 0});
+            }
+        }
+        return {s0.type(), output_dyn_dims};
+    }
+
+    template <class T>
```

Comment:
Can you just use std::size_t, Instead of template?

Reply:
It looks like std::count() returns the iterator difference type, which is probably std::ptrdiff_t. I didn't want to deal with it so I made it a template. Probably could convert it to a size_t since it's going to be >= 0.

Reply:
It should get converted implictly.

Reply:
Did you try removing template ?